### PR TITLE
modified the implement of Lars optimizer

### PR DESCRIPTION
**Pull request by JZ-LIANG**
### PR types
Performance optimization

### PR changes
OPs

### Describe
modified the implement of Lars optimizer, increases it performance in convergence accuracy and convergence speed.
modifications:
* allow to filter parameter of specific layer  from lars weight decay.
* filter all layers that without lars weight decay from lars local lr scaling .

the modification allows the accuracy of resnet50 >= 75.9% when training batch size > 8k.  

|version / batch size | 2k | 8k | 16k |
|:---:|:---:|:---:|:---:|
|origin | 75.34 | 74.28 | 75.4 |
|new | 76.51 | 76.14 | 76.14 |

```python
# origin
optimizer = fluid.optimizer.LarsMomentumOptimizer(
            lars_coeff=0.01, lars_weight_decay=0.0001,  learning_rate = learning_rate,  
            momentum = 0.9, regularization=None)

# new
optimizer = fluid.optimizer.LarsMomentumOptimizer(
            lars_coeff=0.01, lars_weight_decay=0.0001,  learning_rate = learning_rate,  
            momentum = 0.9, regularization=None,
            exclude_from_weight_decay=['batch_norm',"b_0"],  epsilon = 0)
```

### CODE REVIEW

File: python/paddle/distributed/fleet/meta_optimizers/lamb_optimizer.py [hunk 1]
```diff
@@ -84,6 +84,9 @@ def backward(self,
         return self.lamb_opt.backward(loss, startup_program, parameter_list,
                                       no_grad_set, callbacks)
 
+    def apply_gradients(self, params_grads):
```
Comment (gavin1332) [inline python/paddle/distributed/fleet/meta_optimizers/lamb_optimizer.py:95]: comments required to explain why you use `apply_gradient` here.

File: python/paddle/distributed/fleet/meta_optimizers/lars_optimizer.py [hunk 2]
```diff
@@ -68,6 +71,9 @@ def backward(self,
         return self.lars_opt.backward(loss, startup_program, parameter_list,
                                       no_grad_set, callbacks)
 
+    def apply_gradients(self, params_grads):
```
Comment (gavin1332) [inline python/paddle/distributed/fleet/meta_optimizers/lars_optimizer.py:82]: comments required to explain why you use apply_gradient here.

File: paddle/fluid/operators/optimizers/lars_momentum_op.cu [hunk 3]
```diff
@@ -22,15 +22,17 @@ template <typename T>
 __global__ void MomentumLarsKernel(const T* p, const T* g, const T* v,
                                    const T* learning_rate, const T mu,
                                    const int64_t num, const T lars_coeff,
-                                   const T lars_weight_decay, const T* p_norm,
-                                   const T* g_norm, T* p_out, T* v_out) {
+                                   const T lars_weight_decay, const T epsilon,
```
Comment (gavin1332) [inline paddle/fluid/operators/optimizers/lars_momentum_op.cu]: new argument should be push back.

File: python/paddle/fluid/optimizer.py [hunk 4]
```diff
@@ -1651,6 +1653,8 @@ def __init__(self,
                  momentum,
                  lars_coeff=0.001,
                  lars_weight_decay=0.0005,
+                 epsilon=0,
+                 exclude_from_weight_decay=None,
```
Comment (gavin1332) [inline python/paddle/fluid/optimizer.py]: new arguments should be pushed back

File: python/paddle/fluid/optimizer.py [hunk 5]
```diff
@@ -1609,6 +1609,8 @@ class LarsMomentumOptimizer(Optimizer):
             momentum (float): momentum factor
         lars_coeff (float): Defines how much we trust the layer to change its weights.
         lars_weight_decay (float): Weight decay coefficient for decaying using LARS.
+        epsilon (float): Epsilon to avoid Division by Zero when calculate local lr.
+        exclude_from_weight_decay (list[str], optional): Name string of layers which will be exclude from lars weight decay.
```
Comment (gavin1332) [inline python/paddle/fluid/optimizer.py]: new arguments should be pushed back

File: python/paddle/distributed/fleet/meta_optimizers/lamb_optimizer.py (hunk 1 above)
Comment (JZ-LIANG) [reply python/paddle/distributed/fleet/meta_optimizers/lamb_optimizer.py:95]: explained in the code

File: python/paddle/distributed/fleet/meta_optimizers/lars_optimizer.py (hunk 2 above)
Comment (JZ-LIANG) [reply python/paddle/distributed/fleet/meta_optimizers/lars_optimizer.py:82]: explained in the code

File: paddle/fluid/operators/optimizers/lars_momentum_op.cu (hunk 3 above)
Comment (JZ-LIANG) [reply paddle/fluid/operators/optimizers/lars_momentum_op.cu]: modified as required

File: python/paddle/fluid/optimizer.py (hunk 4 above)
Comment (JZ-LIANG) [reply python/paddle/fluid/optimizer.py]: modified as required

File: python/paddle/fluid/optimizer.py (hunk 5 above)
Comment (JZ-LIANG) [reply python/paddle/fluid/optimizer.py]: modified as required

**Review by gavin1332 — DISMISSED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Review by gavin1332 — APPROVED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Review by TCChenlong — APPROVED** [review]
LGTM
